### ISSUE TITLE
FieldAccessContext skips wrappedNode on solveSymbol(...)

### ISSUE THREAD

**Issue by mlangkabel** [ftomassetti/java-symbol-solver#45]
I was just looking at the FieldAccessContext. It doesn't seem to do anything but asking its parent context when solving a symbol. Usually this seems fine except for the following situation:
```
public class Foo{
	private int bar;

	public void test(int bar){
		this.bar = bar;
	}
}
```
In this case the context of the method is asked to solve the name before the class context. So `this.bar` is solved as being the parameter of the `test(...)` method (which is not correct ;) ).

**Comment by mlangkabel** [comment]
I think I've got a crude solution for this:
```
public class FieldAccessContext extends AbstractJavaParserContext<FieldAccessExpr> {

	...

	@Override
	public SymbolReference<? extends ValueDeclaration> solveSymbol(String name, TypeSolver typeSolver) {
		if (wrappedNode.getFieldExpr().toString().equals(name)) {
			Expression scope = wrappedNode.getScope();
			if (scope instanceof ThisExpr) {
				com.github.javaparser.ast.body.TypeDeclaration context = null; 

				Node parentNode = scope.getParentNode();
				while (parentNode != null && !(parentNode instanceof com.github.javaparser.ast.body.TypeDeclaration)) {
					parentNode = parentNode.getParentNode();
				}
				if (parentNode != null) {
					context = (com.github.javaparser.ast.body.TypeDeclaration)parentNode;
				}
				
				if (context != null) { 
					return JavaParserFactory.getContext(context, typeSolver).solveSymbol(name, typeSolver);
				}
			}
		}
		return JavaParserFactory.getContext(wrappedNode.getParentNode(), typeSolver).solveSymbol(name, typeSolver);
	}

	...

}
```

Is there a more elegant way to do this? Right now I don't have a running setup for the project's build and test environment. But I could try to get it running and create a pull request afterwards.

**Comment by mlangkabel** [comment]
solved with merge of pull request #51
